spec/service_worker/index.bs

```diff
@@ -1904,17 +1918,25 @@ spec: webidl; urlPrefix: https://heycam.github.io/webidl/
+                                <li>When <var>promise</var> is fulfilled with an object whose <code>done</code> property is true, <a lt="close ReadableStream">close</a> <var>newStream</var>.
+                                <li>When <var>promise</var> is fulfilled with a value that matches with neither of the above patterns, or <var>promise</var> is rejected, <a lt="error ReadableStream">error</a> <var>newStream</var> with a <code>TypeError</code>.</li>
+                              </ol>
+                            </li>
+                            <li>Let <var>cancel</var> be an action that <a lt="cancel a reader">cancels</a> <var>reader</var>.</li>
+                            <li>Let <var>newStream</var> be the result of <a lt="construct a ReadableStream">constructing</a> a ReadableStream object with <var>strategy</var>, <var>pull</var> and <var>cancel</var>.</li>
+                            <li>Set <var>potentialResponse</var>'s <a>body</a> to a new <a>body</a> whose <a>stream</a> is <var>newStream</var>.</li>
                           </ol>
-                          <p class="note">These substeps are meant to produce the observable equivalent of "piping" response's <a href="https://fetch.spec.whatwg.org/#concept-body-body">body</a>'s <a>stream</a> into potentialResponse. That is, response is left with a <a href="https://fetch.spec.whatwg.org/#concept-body-body">body</a> with a <a>ReadableStream</a> object that is <a>disturbed</a> and <a>locked</a>, while the data readable from potentialResponse's <a>body</a>'s <a>stream</a> is now equal to what used to be response's, if response's original <a href="https://fetch.spec.whatwg.org/#concept-body-body">body</a> is non-null.</p>
-                          <p class="issue">These substeps will be replaced by <a href="https://github.com/slightlyoff/ServiceWorker/issues/850#issuecomment-197893295">using pipe</a> when the algorithm for <a href="https://streams.spec.whatwg.org/#rs-pipe-to">pipeTo</a> becomes stable.</p>
+                          <p class="note">These substeps are meant to produce the observable equivalent of "piping" response's <a href="https://fetch.spec.whatwg.org/#concept-body-body">body</a>'s <a>stream</a> into potentialResponse.</p>
+                          <p class="issue">The user agent doesn't need to preserve buffers' boundary although the above steps require it. These substeps will be replaced by <a href="https://github.com/whatwg/fetch/issues/330#issuecomment-231947405">using pipe</a> when the algorithm for <a href="https://streams.spec.whatwg.org/#rs-pipe-to">pipeTo</a> becomes stable and that will resolve this inconsistency.</p>
```

Comment:
I think it is possible to spec these even without using pipeTo, if you want. When the promise fulfills with a value, transfer it, but then just say something like "store it in _chunkStorage_" instead of enqueueing it. Then, add a step after constructing newStream saying something like "As _chunkStorage_ gains entries, enqueue new Uint8Array objects into _newStream_ in sequence such that their concatenated result is the same as the concatenated result of the values in _chunkStorage_." Maybe add a note explaining, like "This is intended to allow the user agent to arbitrarily change chunk boundaries between _response_'s body's stream and _newStream_."